doc/sg/c_cpp_coding_style.md

```diff
@@ -71,8 +71,8 @@ Example:
 ```c
 // `ptr` can never be NULL for reasons.
 ```
-Documentation comments should start with `/*` and non-doc comments should start with `//`.
-Example:
+
+Documentation comments should use `/* ... */` style blocks. Examples:
```

Comment:
Do we generate documentation for internal code?

If we do, this should specify to use `/** ... */` Doxygen-style comments. I've seen both styles used internally, so unsure whether we require this or not.

Reply:
Good point, would bind change it?

Reply:
Could you also add a text saying something like:
 "structs and enums members should also be documented using `/** ... */` style. "

Reply:
Sure, updated.